src/proxy_handler.cc

```diff
@@ -159,6 +262,33 @@ ProxyHandler::HandleRequest(const Request& request, Response* response)
     if(status_code == 200)
       break;
   }
+  if(cache_.find(request.uri()) != cache_.end()) {
```

Comment:
'while' is for 302 code. 
I think there is no need to cache the 302-responses. For example, there is a url xxx.com request, and it receives 302. After 'while', it receives the 200 response. We should only need to cache this 200 response.